src/Sentry/Internal/ITransactionProfiler.cs

```diff
@@ -0,0 +1,30 @@
+using Sentry.Protocol;
+
+namespace Sentry.Internal;
+
+/// <summary>
+/// Factory to create/attach profilers when a transaction starts.
+/// </summary>
+internal interface ITransactionProfilerFactory
+{
+    /// <summary>
+    /// Called during transaction start to start a new profiler, if applicable.
+    /// </summary>
+    ITransactionProfiler? OnTransactionStart(ITransaction tracer, DateTimeOffset now, CancellationToken cancellationToken);
```

Comment:
Why are we passing `DateTimeOffset` for start/finish methods?  Could the factory create a `Stopwatch` or `SentryStopwatch` internally?

If it would help, I can update `SentryStopwatch` to be stoppable.

Reply:
We need the same stopwatch as the transaction  runs with, otherwise, we may end up with a different timestamp, if the system time changes.

However, after having a second look at `SentryStopwatch`, I see that it's actually not guaranteed to be monotonic anyway 🤔

Reply:
Understood & changed.